### PR TITLE
Include hidden input.pat-autosuggest in validation

### LINKED ISSUE #683: pat-validation ignores "required" on pat-autosuggest
When adding an input.pat-autosuggest with a "required" attribute to a form.pat-validation element, it is ignored and the form can be submitted without entering/selecting a value.

Minimal example:

```
        <form class="pat-validation">
            <input type="text" required="required" name="input1" />
            <input type="text" class="pat-autosuggest" required="required" name="input2" />
            <button type="submit">Submit</button>
        </form>
```

Submitting the form does not work when omitting the first input, but it does work when filling in the first but omitting the second (autosuggest) input.

### PR DESCRIPTION
**Pull request by reinhardt**
Fixes #683

### CODE REVIEW

File: src/pat/validation/tests.js [hunk 1]
```diff
@@ -27,6 +27,24 @@ define(["pat-registry", "pat-validation"], function(registry, pattern) {
...
+            pattern.init($el);
+            $el.submit();
+            expect($el.find('em.warning').length).toBe(1);
+
+            // Check that validation passes when required is not here.
+            $el = $(
+                '<form class="pat-validation">'+
+                '<input type="hidden" name="name" class="pat-autosuggest" data-pat-autosuggest="words: one, two">'+
+                '</form>');
+            $el.submit();
```
Comment (reinhardt) [inline src/pat/validation/tests.js:44]: Do we need another `pattern.init($el)` here?

Comment (thet) [reply src/pat/validation/tests.js:45]: good catch, yes it does. checked it and added it myself.

Comment (reinhardt) [reply src/pat/validation/tests.js:45]: If yes then we probably also need it in the previous test which I copied this from...

Comment (thet) [reply src/pat/validation/tests.js:46]: perfect, also added it there.

**Review by thet — APPROVED** [review]
LGTM

**Pull request merged**